File: tests/algorithms/fibonacci/fibonacci.ts
/**
 * Fibonacci benchmark implementation in TypeScript.
 * Calculates the nth Fibonacci number using recursive algorithm.
 */

/**
 * Calculate the nth Fibonacci number recursively.
 */
function fibonacci(n: number): number {
    if (n <= 1) {
        return n;
    }
    return fibonacci(n - 1) + fibonacci(n - 2);
}

function main(): void {
    // Test parameter
    const n = 35; // Adjusted for reasonable execution time

    console.log(`Calculating fibonacci(${n})...`);
    const startTime = process.hrtime.bigint();

    const result = fibonacci(n);

    const endTime = process.hrtime.bigint();
    const executionTime = Number(endTime - startTime) / 1e9; // Convert to seconds

    console.log(`Result: ${result}`);
    console.log(`Execution time: ${executionTime.toFixed(6)} seconds`);
}

main();

File: tests/algorithms/fibonacci/fibonacci.py
#!/usr/bin/env python3
"""
Fibonacci benchmark implementation in Python.
Calculates the nth Fibonacci number using recursive algorithm.
"""

import sys
import time


def fibonacci(n):
    """Calculate the nth Fibonacci number recursively."""
    if n <= 1:
        return n
    return fibonacci(n-1) + fibonacci(n-2)


def main():
    """Main execution function."""
    # Test parameter
    n = 35  # Adjusted for reasonable execution time
    
    print(f"Calculating fibonacci({n})...")
    start_time = time.perf_counter()
    
    result = fibonacci(n)
    
    end_time = time.perf_counter()
    execution_time = end_time - start_time
    
    print(f"Result: {result}")
    print(f"Execution time: {execution_time:.6f} seconds")


if __name__ == "__main__":
    main()